commcare_connect/opportunity/tasks.py

```diff
@@ -267,3 +269,21 @@ def generate_work_status_export(opportunity_id: int, export_format: str):
     export_tmp_name = f"{now().isoformat()}_{opportunity.name}_payment_verification.{export_format}"
     save_export(dataset, export_tmp_name, export_format)
     return export_tmp_name
+
+
+@celery_app.task()
+def approve_completed_work_and_update_payment_accrued(completed_work_ids: list[int]):
+    """Approves and Updates Payment accrued for all completed (all deliver units submitted and any status) and
+    un-flagged completed work."""
+    for completed_work_id in completed_work_ids:
+        completed_work = CompletedWork.objects.get(pk=completed_work_id)
+        access = completed_work.opportunity_access
+        if completed_work.flags:
+            continue
+        payment_accrued = completed_work.payment_accrued
+        if payment_accrued > 0:
```

Comment:
Even if there is no payment we want to still approve the completed work, right? In case it is necessary for other works, or just because they should show as approved, regardless of value?

On that note, does ordering matter here? So that payment units that depend on each other are approved in the right order?

Reply:
Fixed. https://github.com/dimagi/commcare-connect/pull/294/commits/0d34603ed9262705f8f606e05302fbc577776f21